pkg/controller/dgraph/dgraph.go

```diff
@@ -135,13 +139,27 @@ func ExecuteQuery(query string, root interface{}) error {
 func MutateNode(n []byte) (*api.Assigned, error) {
 	mu := &api.Mutation{
 		CommitNow: true,
+		SetJson:   n,
 	}
 
-	mu.SetJson = n
 	ctx := context.Background()
 	return client.NewTxn().Mutate(ctx, mu)
 }
 
+// DeleteNodes batch deletes given nodes
+func DeleteNodes(n []byte) error {
+	ctx := context.Background()
+	mu := &api.Mutation{
+		CommitNow:  true,
+		DeleteJson: n,
+	}
+	_, err := client.NewTxn().Mutate(ctx, mu)
```

Comment:
We can at least move this part to a separate function called `MutateNode` maybe.